metadata-integration/java/datahub-schematron/lib/src/test/java/io/datahubproject/schematron/converters/avro/AvroSchemaConverterTest.java

```diff
@@ -0,0 +1,942 @@
+        schema.getFields().get(12),
+        "[version=2.0].[type=PrimitiveType].[type=string].nullableStringField",
+        "string",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
+    assertSchemaField(
+        schema.getFields().get(13),
+        "[version=2.0].[type=PrimitiveType].[type=enum].status",
+        "Enum",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new EnumType())));
+  }
+
+  @Test(groups = "basic")
+  void testComplexMaps() throws IOException {
+    SchemaMetadata schema =
+        avroSchemaConverter.toDataHubSchema(
+            readAvroSchema("complex_maps.avsc"), false, false, dataPlatformUrn, null);
+
+    schema.getFields().forEach(System.out::println);
+
+    assertEquals(schema.getFields().size(), 15);
+
+    assertSchemaField(
+        schema.getFields().get(0),
+        "[version=2.0].[type=MapType].[type=map].mapOfString",
+        "map<string,string>",
+        false,
+        false,
+        new SchemaFieldDataType()
+            .setType(
+                SchemaFieldDataType.Type.create(
+                    new MapType().setKeyType("string").setValueType("string"))));
+    assertSchemaField(
+        schema.getFields().get(1),
+        "[version=2.0].[type=MapType].[type=map].[type=ComplexType].mapOfComplexType",
+        "ComplexType",
+        false,
+        false,
+        new SchemaFieldDataType()
+            .setType(
+                SchemaFieldDataType.Type.create(
+                    new MapType().setKeyType("string").setValueType("ComplexType"))));
+    assertSchemaField(
+        schema.getFields().get(2),
+        "[version=2.0].[type=MapType].[type=map].[type=ComplexType].mapOfComplexType.[type=string].field1",
+        "string",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
+    assertSchemaField(
+        schema.getFields().get(3),
+        "[version=2.0].[type=MapType].[type=map].[type=ComplexType].mapOfComplexType.[type=int].field2",
+        "int",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new NumberType())));
+    assertSchemaField(
+        schema.getFields().get(4),
+        "[version=2.0].[type=MapType].[type=map].[type=union].mapOfNullableString",
+        "union",
+        false,
+        false,
+        new SchemaFieldDataType()
+            .setType(
+                SchemaFieldDataType.Type.create(
+                    new MapType().setKeyType("string").setValueType("union"))));
+    assertSchemaField(
+        schema.getFields().get(5),
+        "[version=2.0].[type=MapType].[type=map].[type=union].[type=string].mapOfNullableString",
+        "string",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
```

Comment:
avro schema

```
    {
      "name": "mapOfNullableString",
      "type": {
        "type": "map",
        "values": ["null", "string"]
      }
    },
````

the native type is `union`, no reference to `map`
`expectedNullable` for the value is `false` instead of `true`

it's a `union` but only one tested field 🤔

Reply:
Yeah, it should be nullable, and native type maybe should be union<null, string>